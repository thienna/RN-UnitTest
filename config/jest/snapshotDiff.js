import snapshotDiff, { toMatchDiffSnapshot } from 'snapshot-diff';

expect.extend({ toMatchDiffSnapshot });
expect.addSnapshotSerializer(snapshotDiff.getSnapshotDiffSerializer());
