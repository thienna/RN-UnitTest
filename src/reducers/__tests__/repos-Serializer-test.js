import repos from '../repos';
import { initialState } from '../repos';
import { receiveRepos } from '../../actions/api';
import { repos as data } from '../../../config/jest/mockData';

expect.addSnapshotSerializer({
  test: (val) => {
    return val && val.id
  },
  print: (val) => {
    return `[${val.id}] ${
      val.name}, Description: ${
      val.description}, Stars: ${
      val.stargazers_count}`;
  }
});

it('uses a custom serializer', () => {
  expect(repos(initialState, receiveRepos.success(data))).toMatchSnapshot();
});
