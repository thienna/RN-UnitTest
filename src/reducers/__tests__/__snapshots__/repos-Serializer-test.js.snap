// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uses a custom serializer 1`] = `
Object {
  "error": null,
  "isLoading": false,
  "repos": Array [
    [1] Test 1, Description: Useful description goes here., Stars: 50000,
    [2] Test 2, Description: Useful description goes here., Stars: 10000,
    [3] Test 3, Description: Useful description goes here., Stars: 499,
  ],
  "selected": null,
}
`;
